app/interactors/destroy_order_item.rb

```diff
@@ -0,0 +1,12 @@
+class DestroyOrderItem
+  include Interactor
+
+  def call
+    order_item = OrderItem.find(context.order_item_id)
+    new_total = context.current_order.total_price - order_item.book.price * order_item.quantity
+    order_item.destroy
+    context.current_order.update(total_price: new_total)
```

Comment:
Maybe it would be better to create ServiceObject which will recalculate `total_price` of `order` after add/delete `order_items` to/from `order`?
With a service object, this method might look like this:
```
def call
    order_item = OrderItem.find_by(id: context.order_item_id)
    return context.fail! unless order_item

    order_item.destroy
    UpdateTotalPriceService.call(order, order_item.price, order_item.quantity, :remove)
```

or even better, you can cover these operations into the transaction
```
ActiveRecord::Base.transaction do
    order_item.destroy
    UpdateTotalPriceService.call(order, order_item.price, order_item.quantity, :remove)
end
```
https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html